### ISSUE TITLE
Assertion in row_insert_for_mysql()

### ISSUE THREAD

**Issue by midenok** [tempesta-tech/mariadb#288]
These tests fail on assertion:
```
versioning.rpl_mixed,innodb \
versioning.rpl_row,innodb \
versioning.rpl_stmt,innodb
```
```c++
#3  0x00007ffff5d97fc2 in __GI___assert_fail (assertion=0x1613012 "table->vers_row_start != table->vers_row_end", file=0x16128e0 "/home/midenok/src/mariadb/review/src/storage/innobase/row/row0mysql.cc", line=1475, function=0x1612f53 "dberr_t row_insert_for_mysql(const byte *, row_prebuilt_t *, ins_mode_t)") at assert.c:101
#4  0x0000000000ed092e in row_insert_for_mysql (mysql_rec=0x7fff84026fe8 "\375\001", prebuilt=0x7fff840319c8, ins_mode=ROW_INS_VERSIONED) at /home/midenok/src/mariadb/review/src/storage/innobase/row/row0mysql.cc:1475
#5  0x0000000000d24a35 in ha_innobase::write_row (this=0x7fff84021d38, record=0x7fff84026fe8 "\375\001") at /home/midenok/src/mariadb/review/src/storage/innobase/handler/ha_innodb.cc:8371
#6  0x0000000000a8ed49 in handler::ha_write_row (this=0x7fff84021d38, buf=0x7fff84026fe8 "\375\001") at /home/midenok/src/mariadb/review/src/sql/handler.cc:6050
#7  0x000000000070608a in write_record (thd=0x7fff800018f0, table=0x7fff84030d90, info=0x7ffff021a6d8) at /home/midenok/src/mariadb/review/src/sql/sql_insert.cc:1999
#8  0x00000000007027b6 in mysql_insert (thd=0x7fff800018f0, table_list=0x7fff8000fcc0, fields=..., values_list=..., update_fields=..., update_values=..., duplic=DUP_ERROR, ignore=false) at /home/midenok/src/mariadb/review/src/sql/sql_insert.cc:1060
#9  0x00000000007422c8 in mysql_execute_command (thd=0x7fff800018f0) at /home/midenok/src/mariadb/review/src/sql/sql_parse.cc:4693
#10 0x0000000000739e4f in mysql_parse (thd=0x7fff800018f0, rawbuf=0x7fff8001e8db "insert into t1 values (1)", length=25, parser_state=0x7ffff021cee0, is_com_multi=false, is_next_command=false) at /home/midenok/src/mariadb/review/src/sql/sql_parse.cc:7947
```
```c++
(gdb) f 4
#4  0x0000000000ed092e in row_insert_for_mysql (mysql_rec=0x7fff84026fe8 "\375\001", prebuilt=0x7fff840319c8, ins_mode=ROW_INS_VERSIONED) at /home/midenok/src/mariadb/review/src/storage/innobase/row/row0mysql.cc:1475
1475                    ut_ad(table->vers_row_start != table->vers_row_end);
```
```c++
1467            row_get_prebuilt_insert_row(prebuilt);
1468            node = prebuilt->ins_node;
1469
1470            row_mysql_convert_row_to_innobase(node->row, prebuilt, mysql_rec,
1471                                              &blob_heap);
1472
1473            if (ins_mode != ROW_INS_NORMAL)
1474            {
1475                    ut_ad(table->vers_row_start != table->vers_row_end);
1476                    /* Return back modified fields into mysql_rec, so that
1477                       upper logic may benefit from it (f.ex. 'on duplicate key'). */
1478                    const mysql_row_templ_t* t = &prebuilt->mysql_template[table->vers_row_end];
1479                    ut_ad(t->mysql_col_len == 8);
1480
1481                    if (ins_mode == ROW_INS_HISTORICAL) {
1482                            set_tuple_col_8(node->row, table->vers_row_end, trx->id, node->entry_sys_heap);
1483                    }
1484                    else /* ROW_INS_VERSIONED */ {
1485                            set_tuple_col_8(node->row, table->vers_row_end, IB_UINT64_MAX, node->entry_sys_heap);
1486                            int8store(&mysql_rec[t->mysql_col_offset], IB_UINT64_MAX);
1487                            t = &prebuilt->mysql_template[table->vers_row_start];
1488                            ut_ad(t->mysql_col_len == 8);
1489                            set_tuple_col_8(node->row, table->vers_row_start, trx->id, node->entry_sys_heap);
```

**Comment by midenok** [comment]
Related to #287

**Comment by kevgs** [comment]
Fixed along with #290.
